### PR TITLE
Build redpanda from sources

### PR DESCRIPTION
**Pull request by layus**
- rename redpanda -> redpanda-bin
- Add redpanda built from sources

### CODE REVIEW

**Comment by layus** [comment]
I thought I had everything fixed, but redpanda from sources coredumps in at least one of the tests :grimacing: .

So I will need to investigate more.

**Comment by layus** [comment]
@kfollesdal @Radvendii I would like to get a review from both of you because I change the package names

redpanda -> redpanda-bin
redpanda-console -> redpanda-console-bin

And some new packages:
* redpanda-{server,client}: build from source
* redpanda-{server,client}-bin: aliases to redpanda-bin

**Review by Radvendii — APPROVED** [review]
LGTM. only a few minor things

File: modules/redpanda.nix [hunk 1]
```diff
@@ -278,13 +278,16 @@ in
...
+        # XXX: recent redpanda versions do not need it.
+        # They rely on bundled hwloc-xxx-redpanda binaries
         pkgs.hwloc
         pkgs.util-linux
         pkgs.inetutils
         pkgs.gawk
       ];
       script = ''
         set -euo pipefail
+        set -x
```
[the remark this answers is not in the document]
Comment (layus) [reply modules/redpanda.nix:290]: Oops, leftover debugging

File: flake.nix [hunk 2]
```diff
@@ -14,46 +14,41 @@
...
           imports = [ ./modules/redpanda-console.nix ];
           # FIXME: once we have a redpanda-console in nixpkgs, this can be removed
-          services.redpanda-console.package = pkgs.callPackage ./packages/redpanda-console { };
+          services.redpanda-console.package = lib.mkDefault (pkgs.callPackages ./packages { }).redpanda-console;
         };
         redpanda-acl = import ./modules/redpanda-acl.nix;
       };
 
-      packages.${system} = { inherit (pkgs) redpanda redpanda-console; };
+      packages.${system} = import ./packages { inherit (pkgs) callPackage callPackages; };
```
[the remark this answers is not in the document]
Comment (layus) [reply flake.nix:45]: Oh, right, makes sense.

File: flake.nix [hunk 3]
```diff
@@ -14,46 +14,41 @@
       system = "x86_64-linux";
       pkgs = import nixpkgs {
         inherit system; config.allowUnfree = true;
-        overlays = builtins.attrValues self.overlays;
```
Comment (kfollesdal) [inline flake.nix:17]: When removing overlay, the redpanda in devshell is referencing to broken nixpkgs version. Maybe remove redpanda form devshell? Probably do not need it there

**Review by kfollesdal — APPROVED** [review]
Good work 🎉  One comment about overlay and redpanda version in devshell.

Agree with the `-bin` naming. First i did not see the reason for aliasing `redpanda-client-bin ` and `redpanda-server-bin` to `redpanda-bin`, but hen I see that the source build make a distinction between client and server. So then the aliasing make sanse for completness :-)

**Pull request merged**